src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -0,0 +1,418 @@
+    }
+
+    return integrationLogTypeIsDefinedBool && integrationStepperTypeIsDefinedBool && timeStep_.isDefined() && relativeTolerance_.isDefined() && absoluteTolerance_.isDefined();
+}
+
+void                            NumericalSolver::print                      (           std::ostream&               anOutputStream,
+                                                                                        bool                        displayDecorator                            ) const
+{
+
+    displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "NumericalSolver") : void () ;
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration logging type:"                << NumericalSolver::StringFromIntegrationLogType(integrationLogType_) ;
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration stepper type:"                << NumericalSolver::StringFromIntegrationStepperType(integrationStepperType_) ;
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration time step:"                << (timeStep_.isDefined() ? timeStep_.toString() : "Undefined") ;
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration relative tolerance:"                << (relativeTolerance_.isDefined() ? relativeTolerance_.toString() : "Undefined") ;
+    ostk::core::utils::Print::Line(anOutputStream) << "Integration absolute tolerance:"                << (absoluteTolerance_.isDefined() ? absoluteTolerance_.toString() : "Undefined") ;
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void () ;
+
+}
+
+NumericalSolver::IntegrationLogType  NumericalSolver::getIntegrationLogType ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return integrationLogType_ ;
+
+}
+
+NumericalSolver::IntegrationStepperType  NumericalSolver::getIntegrationStepperType ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return integrationStepperType_ ;
+
+}
+
+Real                            NumericalSolver::getTimeStep                ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return timeStep_ ;
+
+}
+
+Real                            NumericalSolver::getRelativeTolerance       ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return relativeTolerance_ ;
+
+}
+
+Real                            NumericalSolver::getAbsoluteTolerance       ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("NumericalSolver") ;
+    }
+
+    return absoluteTolerance_ ;
+
+}
+
+NumericalSolver::StateVector    NumericalSolver::integrateStateForDuration  (   const   StateVector&                anInitialStateVector,
```

Comment:
Might be worth leveraging `integrate_times` as well in a integrateStateForDurations. Atleast worth benchmarking the efficiency of N `integrate_adaptive` vs 1 `integrate_times` for N timesteps

Reply:
integrate_times offers less flexibility for the integrator to determine its own appropriate it seems? I might be wrong. Have you found that is offers better performance than integrate_adaptive?